frontend/src/components/ReopenReasonModal.js

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { REOPEN_REASONS, GOAL_STATUS } from '@ttahub/common';
+import {
+  Form, FormGroup, ErrorMessage, Label, Fieldset, Radio, Textarea,
+} from '@trussworks/react-uswds';
+import Modal from './Modal';
+
+const ReopenReasonModal = ({
+  modalRef, goalId, onSubmit, resetValues,
+}) => {
+  const [reopenReason, setReopenReason] = useState('');
+  const [reopenContext, setReopenContext] = useState('');
+  const [showValidationError, setShowValidationError] = useState(false);
+
+  useEffect(() => {
+    // Every time we show the modal reset the form.
+    setReopenReason('');
+    setReopenContext('');
+    setShowValidationError(false);
+  }, [resetValues]);
+
+  const reasonChanged = (e) => {
+    setReopenReason(e.target.value);
+    setShowValidationError(false);
+  };
+
+  const reasonRadioOptions = Object.values(REOPEN_REASONS[GOAL_STATUS.CLOSED]);
+
+  const generateReasonRadioButtons = () => reasonRadioOptions.map((r) => (
+    <Radio
+      id={r.trim().replace(' ', '-').toLowerCase()}
+      key={r}
+      onChange={reasonChanged}
+      name="reopenReason"
+      label={r}
+      value={r}
+      className="smart-hub--report-checkbox"
+      checked={reopenReason === r}
+    />
+  ));
+  const contextChanged = (e) => {
+    setReopenContext(e.target.value);
+  };
+
+  const validateSubmit = () => {
+    if (!reopenReason) {
+      setShowValidationError(true);
+    } else {
+      onSubmit(goalId, reopenReason, reopenContext);
+    }
+  };
+
+  return (
+    <div className="smart-hub--goal-close-suspend-reason">
+      <Modal
+        modalRef={modalRef}
+        onOk={validateSubmit}
+        modalId="ReopenReasonModal"
+        title="Why are you reopening this goal?"
+        okButtonText="Submit"
+        okButtonAriaLabel="Reopen goal"
+        okButtonCss="usa-button--primary"
+        cancelButtonCss="usa-button--unstyled"
+        showTitleRequired
+        forceAction
+      >
+        <Form
+          key={`reopen-reason-form-goal-${goalId}`}
+        >
+          <FormGroup error={showValidationError} className="margin-top-0">
+            <Fieldset>
+              {showValidationError
+                ? (
+                  <ErrorMessage>
+                    Please select a reason for reopening this goal.
+                  </ErrorMessage>
+                ) : null}
+              {
+                generateReasonRadioButtons()
+              }
+            </Fieldset>
+          </FormGroup>
```

Comment:
```suggestion
          <FormGroup error={showValidationError} className="margin-top-0">
            <ErrorMessage>
              {showValidationError
                ? 'Please select a reason for reopening this goal.' : null}
            </ErrorMessage>
            <Fieldset>
              {
                generateReasonRadioButtons()
              }
            </Fieldset>
          </FormGroup>
```
I think we have to revisit all the modals with a screenreader test, which I will log a ticket for. But just concentrating on this one: role=alert only announces when content is updated, not created, so I think we need to prime the browser by having an empty span until we need it. ([docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role))

Even with that change, nothing was being announced until I moved the message out of the ```<Fieldset>```. The suggested change gets the validation announced correctly. Maybe verify it still looks right as well? Does to me.